multy_test/test_properties.cpp

```diff
@@ -0,0 +1,385 @@
+    BinaryDataPtr binaty_data_property1(make_clone(BinaryData{data_vals1, 4}));
+    BinaryDataPtr binaty_data_property2(make_clone(BinaryData{data_vals2, 2}));
+
+    properties.bind_property("v", &binaty_data_property1);
+
+    error.reset(properties_set_binary_data_value(&properties, "v", binaty_data_property2.get()));
+    EXPECT_EQ(nullptr, error);
+    EXPECT_EQ(*binary_data_standart , *binaty_data_property1);
+    EXPECT_NE(nullptr, &properties);
+}
+
+GTEST_TEST(PropertiesTestValidArgs, set_public_key_value)
+{
+    Properties properties("");
+    PublicKeyPtr pub_key;
+    PublicKeyPtr pub_key1;
+
+    AccountPtr account;
+    ErrorPtr error(
+            make_account(
+                    CURRENCY_BITCOIN,
+                    "cQeGKosJjWPn9GkB7QmvmotmBbVg1hm8UjdN6yLXEWZ5HAcRwam7",
+                    reset_sp(account)));
+    ASSERT_EQ(nullptr, error);
+    ASSERT_NE(nullptr, account);
+    EXPECT_EQ("mzqiDnETWkunRDZxjUQ34JzN1LDevh5DpU", account->get_address());
+    AccountPtr account1;
+    error.reset( make_account(
+                    CURRENCY_BITCOIN,
+                    "cScuLx5taDyuAfCnin5WWZz65yGCHMuuaFv6mgearmqAHC4p53sz",
+                    reset_sp(account1)));
+    ASSERT_EQ(nullptr, error);
+    ASSERT_NE(nullptr, account);
+    EXPECT_EQ("mfgq7S1Va1GREFgN66MVoxX35X6juKov6A", account1->get_address());
+
+    pub_key =  account->get_public_key();
+    pub_key1 = account1->get_public_key();
+
+    properties.bind_property("v", &pub_key);
+
+    error.reset(properties_set_public_key_value(&properties, "v", pub_key1.get()));
+    EXPECT_EQ(nullptr, error);
+    EXPECT_EQ(*pub_key, *pub_key1);
+    EXPECT_NE(nullptr, &properties);
+
+
+}
+
+GTEST_TEST(PropertiesTestValidArgs, reset_value)
+{
+    ErrorPtr error;
+    Properties properties("");
+    int32_t int_property = 1;
+    properties.bind_property("v", &int_property);
+
+    error.reset(properties_reset_value(&properties, "v"));
+    EXPECT_EQ(nullptr, error);
+    EXPECT_EQ(1, int_property);
+}
+
+GTEST_TEST(PropertiesTestValidArgs, validate)
+{
+    ErrorPtr error;
+    Properties properties("");
+    int32_t int_property = 1;
+
+    error.reset(properties_validate(&properties));
+    EXPECT_EQ(nullptr, error);
+
+    properties.bind_property("v", &int_property);
+    properties.set_property("v", 5);
+
+    error.reset(properties_validate(&properties));
+    EXPECT_EQ(nullptr, error);
+}
+
+GTEST_TEST(PropertiesTestValidArgs, get_specification)
+{
+    // что должен делать и возвращать этот метод
+    ErrorPtr error;
+    Properties properties("");
+    ConstCharPtr specification;
+    int int_property = 3;
+
+    error.reset(properties_get_specification(&properties, reset_sp(specification)));
+    EXPECT_EQ(nullptr, error);
+
+    properties.bind_property("v", &int_property);
+    properties.set_property("v", 5);
+
+    error.reset(properties_get_specification(&properties, reset_sp(specification)));
```

Comment:
Please check that specification is non-empty and contains a bound property name. Also please make a property name really unique here, something like "FOO_BAR_PROPERTY_NAME_FOO_BAR".